documentation/project-docs/repro-helix-failure.md

```diff
@@ -0,0 +1,65 @@
+# Repro Helix Failure
+
+## General
+
+Most of the test failure on helix should be able to repro locally. Try follow [developer-guide](developer-guide.md) _Run tests from the command line_ session first.
+
+## Repro test as tool layout
+
+Run test on Helix reuse "test as tool" infrastructure. All tests projects are published with xunit command line. The SDK under test, the published test project and other dependencies are copied to a helix machine. This step changed the repo structure. Most of the helix-only failures are caused by tests depending on the repo structure or having dependencies that are not carried by the test project publishing step. The following steps helps you to repro the test layout as in the helix machine. And you can run the individual test to discover the cause.
+
+### repro the test layout as in the helix machine
+
+```cmd
+REM fully build first
+.\build.cmd
+
+REM use build env to have stage 0 dotnet on the PATH
+.\artifacts\sdk-build-env.bat
+
+REM run the special test target CreateLocalHelixTestLayout. To have the test layout created on disk.
+dotnet msbuild /restore /t:CreateLocalHelixTestLayout .\src\Tests\UnitTests.proj /p:creator=dotnetsdkdev  /p:_CustomHelixTargetQueue=Windows.Server.Amd64.VS2019.Pre.Open /bl
+```
+
+Copy the result of `artifacts\bin\localHelixTestLayout` to another directory or VM. For example to `C:\helix\localHelixTestLayout`. See "Folders under localHelixTestLayout" for the content. This is _correlation_ payload in helix term.
+
+Publish the test project you want to repro. For example Microsoft.NET.Build.Tests.
+
+```cmd
+dotnet publish src\Tests\Microsoft.NET.Build.Tests\
+```
+
+The output is in `artifacts\bin\Tests\Microsoft.NET.Build.Tests\Debug\publish`. Copy this folder to another directory or VM. For example `C:\helix\payload-dir`. This is the "workitem payload" in helix term.
+
+On the other machine or directory
+
+```cmd
+cd C:\helix\payload-dir
+
+REM HELIX_CORRELATION_PAYLOAD would be set to correlation payload by real helix machine
+set HELIX_CORRELATION_PAYLOAD=C:\helix\localHelixTestLayout
+
+REM "true" is full framework test. Without "true", it is dotnet core tests. RunTestsOnHelix.cmd is the same script will setup the helix environnement.
+C:\helix\localHelixTestLayout\t\RunTestsOnHelix.cmd true
+```
+
+Example of running a specific method. "Microsoft.NET.Build.Tests.dll" need to match the test project target output dll.
+
+```cmd
+dotnet Microsoft.NET.Build.Tests.dll -testExecutionDirectory %TestExecutionDirectory% -msbuildAdditionalSdkResolverFolder %HELIX_CORRELATION_PAYLOAD%\r -html testResults.html -method "Microsoft.NET.Build.Tests.GivenThatWeWantToBuildADesktopExeWithFSharp.It_builds_a_simple_net50_app"
+```
+
+## Folders under localHelixTestLayout
+
+Due to Helix long path problem. The folders under localHelixTestLayout all have one letter name.
```

Comment:
Can we enable long paths on Helix machines?  We should try to avoid having to workaround long path issues, as we're likely to eventually go over the max path and have hard-to-diagnose issues.

Reply:
I had a long talk about the long path issue. Last year for at least a month. (We first found it 2 years ago for the perf tests and started to ask for it) it is not easy, due to the way it is configured. Although, we get the path on helix trimmed, it used to have 2 guid in front.